enterprise/server/cmd/ci_runner/main.go

```diff
@@ -433,12 +532,12 @@ func (ar *actionRunner) Run(ctx context.Context, ws *workspace, startTime time.T
 	}
 
 	// Flush whenever the log buffer fills past a certain threshold.
-	ar.log.writeListener = func() {
-		if size := ar.log.Len(); size >= progressFlushThresholdBytes {
-			ar.flushProgress() // ignore error; it will surface in `bep.Wait()`
+	ar.reporter.log.writeListener = func() {
+		if size := ar.reporter.log.Len(); size >= progressFlushThresholdBytes {
+			ar.reporter.FlushProgress() // ignore error; it will surface in `bep.Wait()`
 		}
```

Comment:
could this logic move inside the reporter class?

Reply:
+1, might make sense to change `PublishStarted` to `Start`, move this logic inside `Start` (after publishing the Started event), and require a symmetric call to `Stop` which stops the flush?

The current logic makes it so that we publish Started, then WorkspaceStatus, and only then do we start flushing progress, but I think it's fine to start flushing progress as soon as we publish the Started event (Bazel does this anyway -- sometimes it even sends progress _before_ publishing the Started event but I think we are able to handle that properly as well, up to a certain number of events)

Reply:
done